### PR TITLE
models for "rent apartments and houses"

### PR DESCRIPTION
**Pull request by liubamiron**

### CODE REVIEW

File: address.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Address extends Model
```
Comment (ghost) [inline address.php]: The file name should be `Address` as your class name

File: benefit.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Benefit extends Model
```
Comment (ghost) [inline benefit.php:8]: The file name should be `Benefit` as your class name

File: customer.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Customer extends Model
```
Comment (ghost) [inline customer.php:8]: The file name should be `Customer` as your class name

File: imobil.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Imobil extends Model
```
Comment (ghost) [inline imobil.php:8]: The file name should be `Imobil` as your class name

Comment (ghost) [reply imobil.php:8]: also `imobil` is a Romanian word, please try to find an English one. I propose `Property`, these changes imply renaming your table

File: imobil.php [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    public function address()
+    {
+        return $this->belongsTo(Address::class);
+    }
+    public function owner()
+    {
+        return $this->belongsTo(Owner::class);
+    }
+
+    public function benefit()
```
Comment (ghost) [inline imobil.php:35]: if hasMany, then method name should be on plural `benefits`

File: imobil.php [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    public function order()
+    {
+        return $this->hasOne(Order::class);
+    }
+
+
+}
+/*  CONSTRAINT `imobils_ibfk_1`
```
Comment (ghost) [inline imobil.php:47]: please remove this comment

File: order.php [hunk 7]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Order extends Model
```
Comment (ghost) [inline order.php:8]: The file name should be `Order` as your class name

File: owner.php [hunk 8]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Owner extends Model
```
Comment (ghost) [inline owner.php:8]: The file name should be `Owner` as your class name

File: imobil.php [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        'bath',
+        'balcony',
+        'description',
+        'created_at',
+    ];
+
+
+    public function address()
+    {
+        return $this->belongsTo(Address::class);
```
Comment (ghost) [inline imobil.php:28]: here you should use `hasOne` and in address `belongsTo`

File: imobil.php [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    }
+    public function owner()
+    {
+        return $this->belongsTo(Owner::class);
+    }
+
+    public function benefit()
+    {
+        return $this->hasMany(Benefit::class);
```
Comment (ghost) [inline imobil.php:37]: this is a relation many to many
https://laravel.com/docs/8.x/eloquent-relationships#many-to-many-model-structure

File: customer.php [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+    protected $fillable = [
+        'name',
+        'email',
+        'telephone',
+        'created_at',
+    ];
+
+    public function order()
+    {
+        return $this->hasOne(Order::class);
```
Comment (ghost) [inline customer.php:20]: One customer can order one or more imobils?
If one, leave as it is.
If no, transform in one to many

File: benefit.php [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+{
+    use HasFactory;
+    protected $fillable = [
+        'name',
+    ];
+
+    
+    
+    public function imobil()
```
Comment (ghost) [inline benefit.php:17]: you should use on plural, because you have many imobils on one benefit

File: address.php [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    'house_nr',
+    'apart_nr',
+    'city',
+    ];
+    
+    public function imobil()
+    {
+       
+    
+    return $this->hasOne(Imobil::class);
```
Comment (ghost) [inline address.php]: in your case address belongs to imobil

**Review by ghost — CHANGES_REQUESTED** [review]

File: address.php (hunk 13 above)
Comment (liubamiron) [reply address.php]: change into return $this->belongsTo(Imobil::class); - deoarce este relatie inversa la one-to-one

File: customer.php (hunk 3 above)
Comment (liubamiron) [reply customer.php:8]: change into Customer.php

File: customer.php (hunk 11 above)
Comment (liubamiron) [reply customer.php:20]: you are right, one customer can order many imobils
        public function orders() {
        return $this->hasMany(Order::class);
    }

File: address.php (hunk 1 above)
Comment (liubamiron) [reply address.php]: change into Address.php

File: address.php (hunk 13 above)
Comment (liubamiron) [reply address.php]: tabela addresses are FOREIGN KEY (`imobil_id`) si este o relatie inversa a unu la unu ( si trebuia sa scriu 
return $this->belongsTo(Imobil::class))

File: imobil.php (hunk 4 above)
Comment (liubamiron) [reply imobil.php:8]: change into Imobil.php

File: imobil.php (hunk 6 above)
Comment (liubamiron) [reply imobil.php:47]: removed the comment

File: imobil.php (hunk 9 above)
Comment (liubamiron) [reply imobil.php:28]: public function address()
    {
        return $this->hasOne(Address::class);
    }

File: imobil.php (hunk 5 above)
Comment (liubamiron) [reply imobil.php:35]: public function benefits()
    {
        return $this->belongsToMany(Benefit::class);
    }

**Pull request closed without merging**